SocialCareCaseViewerApi.Tests/V1/UseCase/PersonalRelationships/PersonalRelationshipsExecutePostUseCaseTests.cs

```diff
@@ -0,0 +1,156 @@
+
+            _person = TestHelpers.CreatePerson((int) _request.PersonId);
+            _otherPerson = TestHelpers.CreatePerson((int) _request.OtherPersonId);
+
+            var personalRelationship = PersonalRelationshipsHelper.CreatePersonalRelationship(_person, _otherPerson, typeInExistingRelationship);
+            _person.PersonalRelationships = new List<PersonalRelationship>() { personalRelationship };
+
+            _mockDatabaseGateway.Setup(x => x.GetPersonsByListOfIds(It.IsAny<List<long>>()))
+                .Returns(new List<Person>() { _person, _otherPerson });
+
+            _mockDatabaseGateway.Setup(x => x.GetPersonWithPersonalRelationshipsByPersonId(It.IsAny<long>(), It.IsAny<bool>()))
+                .Returns(_person);
+
+            _mockDatabaseGateway.Setup(x => x.CreatePersonalRelationship(It.IsAny<CreatePersonalRelationshipRequest>()))
+                .Returns(personalRelationship);
+        }
+
+        [Test]
+        public void CallsDatabaseGatewayToCheckPersonAndOtherPersonExists()
+        {
+            _personalRelationshipsUseCase.ExecutePost(_request);
+
+            _mockDatabaseGateway.Verify(gateway => gateway.GetPersonsByListOfIds(new List<long>() { _request.PersonId, _request.OtherPersonId }));
+        }
+
+        [Test]
+        public void WhenPersonDoesNotExistThrowsPersonNotFoundException()
+        {
+            _mockDatabaseGateway.Setup(x => x.GetPersonsByListOfIds(It.IsAny<List<long>>()))
+                .Returns(new List<Person>() { new Person() { Id = _request.OtherPersonId } });
+
+            Action act = () => _personalRelationshipsUseCase.ExecutePost(_request);
+
+            act.Should().Throw<PersonNotFoundException>()
+                .WithMessage($"\"personId\" with \"{_request.PersonId}\" was not found.");
+        }
+
+        [Test]
+        public void WhenOtherPersonDoesNotExistThrowsPersonNotFoundException()
+        {
+            _mockDatabaseGateway.Setup(x => x.GetPersonsByListOfIds(It.IsAny<List<long>>()))
+                .Returns(new List<Person>() { new Person() { Id = _request.PersonId } });
+
+            Action act = () => _personalRelationshipsUseCase.ExecutePost(_request);
+
+            act.Should().Throw<PersonNotFoundException>()
+                .WithMessage($"\"otherPersonId\" with \"{_request.OtherPersonId}\" was not found.");
+        }
+
+        [Test]
+        public void CallsDatabaseGatewayToCheckTypeExists()
+        {
+            _personalRelationshipsUseCase.ExecutePost(_request);
+
+            _mockDatabaseGateway.Verify(gateway => gateway.GetPersonalRelationshipTypeByDescription(_request.Type));
+        }
+
+        [Test]
+        public void WhenTypeDoesNotExistThrowsPersonalRelationshipTypeNotFoundException()
+        {
+            _mockDatabaseGateway.Setup(x => x.GetPersonalRelationshipTypeByDescription(It.IsAny<string>()))
+                .Returns((PersonalRelationshipType) null);
```

Comment:
If you want your mock to return null you can just leave off the `.Returns(...)`

Reply:
Ahhhh that's true! But I think I'd prefer to keep it and have the test be explicit about it for readability and clarity.